indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/SinglePhaseSubTask.java

```diff
@@ -424,4 +479,170 @@ public Granularity getSegmentGranularity()
       }
     }
   }
+
+  @GET
+  @Path("/unparseableEvents")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getUnparseableEvents(
+      @Context final HttpServletRequest req,
+      @QueryParam("full") String full
+  )
+  {
+    IndexTaskUtils.datasourceAuthorizationCheck(req, Action.READ, getDataSource(), authorizerMapper);
```

Comment:
I don't think this needs to change in this PR, but be aware that there has been some discussion/activity on moving ingestion APIs to use WRITE instead of READ, see #11680.